appsTransport.go

```diff
@@ -106,16 +139,101 @@ func (t *AppsTransport) RoundTrip(req *http.Request) (*http.Response, error) {
+// Specify the AppID of the GitHub App. Either this or ClientID must be
+// specified, but ClientID is now preferred.
+func WithAppID(appID int64) AppsTransportOption {
+	return func(at *AppsTransport) {
+		at.appID = appID
+		at.issuer = strconv.FormatInt(appID, 10)
+	}
+}
+
+// WithClientID specifies the ClientID of the GitHub App. Either this or AppID
+// must be specified, but ClientID is now preferred.
+func WithClientID(clientID string) AppsTransportOption {
+	return func(at *AppsTransport) {
+		at.issuer = clientID
+	}
+}
+
+// WithSigner configures the AppsTransport to use the given Signer for
+// generating JWT tokens.
 func WithSigner(signer Signer) AppsTransportOption {
 	return func(at *AppsTransport) {
-		at.signer = signer
+		at.signerFunc = func() (Signer, error) { return signer, nil }
+	}
+}
+
+// WithPrivateKeyFile directs the transport to use the specified private key
+// file for signing requests. The private key is loaded and processed as an RSA
+// key in PEM format as provided by GitHub in the application settings.
+func WithPrivateKeyFile(privateKeyFile string) AppsTransportOption {
+	return func(at *AppsTransport) {
+		at.signerFunc = currySignerFunc(createSignerFromFile, privateKeyFile)
+	}
+}
+
+// WithPrivateKeyRaw directs the transport to use the specified private key for
+// signing requests. The private key is loaded and processed as an RSA key in
+// PEM format as provided by GitHub in the application settings.
+func WithPrivateKeyRaw(key []byte) AppsTransportOption {
+	return func(at *AppsTransport) {
+		at.signerFunc = currySignerFunc(createSignerFromBytes, key)
+	}
+}
+
+// WithPrivateKeyRaw directs the transport to use the specified private key for
+// signing requests. The private key is provided by GitHub in the application
+// settings.
+func WithPrivateKey(key *rsa.PrivateKey) AppsTransportOption {
+	return func(at *AppsTransport) {
+		at.signerFunc = currySignerFunc(createSignerFromKey, key)
+	}
+}
+
+func createSignerFromFile(privateKeyFile string) (Signer, error) {
+	privateKey, err := os.ReadFile(privateKeyFile)
+	if err != nil {
+		return nil, fmt.Errorf("could not read private key: %s", err)
+	}
+
+	return createSignerFromBytes(privateKey)
+}
+
+func createSignerFromBytes(rawKey []byte) (Signer, error) {
+	key, err := jwt.ParseRSAPrivateKeyFromPEM(rawKey)
+	if err != nil {
+		return nil, fmt.Errorf("could not parse private key: %w", err)
+	}
+
+	return createSignerFromKey(key)
+}
+
+func createSignerFromKey(key *rsa.PrivateKey) (Signer, error) {
+	return NewRSASigner(jwt.SigningMethodRS256, key), nil
+}
+
+// currySignerFunc allows the createSigner* functions to be simple, callable
+// functions that can be used directly, while still supporting the function
+// interface required for config options.
+func currySignerFunc[T any](deferred func(T) (Signer, error), arg T) signerFunc {
+	return func() (Signer, error) {
+		return deferred(arg)
 	}
```

Comment:
I'm aware this may raise some eyebrows. While they're hidden in the hairline, let me explain 😀 

This is roughly equivalent to 

```go

type SignerFunc[T any] struct {
	F func(T) (Signer, error)
	T T
}

func (sf SignerFunc[T]) Create() (Signer, error) {
	return sf.F(sf.T)
}
```

Without this, each of the `createSigner*` methods would have to return a `func() (Signer, error)` instead, and calling from `createSignerFromFile` to `createSignerFromBytes` would need to be written as `createSignerFromBytes(privateKey)()`.

All this said, my goal here is to write something others are happy to maintain, so I'm very open to changing this.